modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp

```diff
@@ -85,6 +112,36 @@ namespace
             cv::resize(in, out, sz, fx, fy, interp);
         }
     };
+
+    G_TYPED_KERNEL(GSubstractBackground, <GMat(GMat)>, "org.opencv.test.substract_background")
+    {
+         static GMatDesc outMeta(GMatDesc in) { return in.withType(CV_8U, 1); }
+    };
+
+    GAPI_OCV_KERNEL_ST(GOCVSubstractBackground, GSubstractBackground, cv::BackgroundSubtractor)
+    {
+        static void setup(const cv::GMatDesc &/* desc */,
+                          const cv::GCompileArgs &compileArgs,
+                          std::shared_ptr<BackgroundSubtractor> &state)
+        {
+            auto sbParams = getCompileArg<SubstractBackgroundStateParams>(compileArgs)
+                                .value_or(SubstractBackgroundStateParams { });
+
+            Ptr<BackgroundSubtractor> model;
+            if (sbParams.method == "knn")
+                model = createBackgroundSubtractorKNN();
+            else if (sbParams.method == "mog2")
```

Comment:
I know it is here for testing purposes only, but in this particular case, it could be an enum defined directly as operation argument at `G_API_OP` level.

We don't put the resize interpolation type as a `compile_arg` :)

Reply:
Yes, agree with you!